test/system/src/test/java/io/pravega/test/system/StreamCutsTest.java

```diff
@@ -0,0 +1,279 @@
+        log.debug("Finished writing events to streams.");
+
+        // Second, get StreamCuts for each slice from both Streams at the same time (may be different in each execution).
+        List<Map<Stream, StreamCut>> streamSlices = getStreamCutSlices(clientFactory, readerGroup);
+        log.debug("Finished creating StreamCuts.");
+
+        // Third, ensure that reader groups can correctly read slices from different Streams.
+        int readEvents, groupId = 0;
+        Map<Stream, StreamCut> startingPoint = null;
+        final int parallelSegments = PARALLELISM_ONE + PARALLELISM_TWO;
+        ReaderGroupConfig.ReaderGroupConfigBuilder configBuilder = ReaderGroupConfig.builder()
+                                                                                    .stream(Stream.of(SCOPE, STREAM_ONE))
+                                                                                    .stream(Stream.of(SCOPE, STREAM_TWO));
+        for (Map<Stream, StreamCut> endingPoint : streamSlices) {
+            configBuilder = configBuilder.endingStreamCuts(endingPoint);
+            if (startingPoint != null) {
+                configBuilder = configBuilder.startingStreamCuts(startingPoint);
+            }
+
+            // Create a new reader group per stream cut slice and read in parallel only events within the cut.
+            final String readerGroupId = READER_GROUP + String.valueOf(groupId);
+            readerGroupManager.createReaderGroup(readerGroupId, configBuilder.build());
+            readEvents = readDummyEvents(clientFactory, readerGroupId, parallelSegments).stream()
+                                                                                        .map(CompletableFuture::join)
+                                                                                        .reduce((a, b) -> a + b).get();
+            log.debug("Read events by group {}: {}.", readerGroupId, readEvents);
+            assertEquals("Expected events read: ", CUT_SIZE, readEvents);
+            startingPoint = endingPoint;
+            groupId++;
+        }
+
+        // Test that a reader group can be reset correctly.
+        readerGroup.resetReaderGroup(configBuilder.startingStreamCuts(streamSlices.get(0)).build());
+        log.debug("Resetting existing reader group {} to stream cut {}.", READER_GROUP, streamSlices.get(0));
+        readEvents = readDummyEvents(clientFactory, readerGroup.getGroupName(),
+                parallelSegments).stream().map(CompletableFuture::join).reduce((a, b) -> a + b).get();
+        assertEquals("Expected read events: ", TOTAL_EVENTS - CUT_SIZE, readEvents);
+        log.debug("All events correctly read from StreamCut slices on multiple Streams. StreamCuts test passed.");
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+    }
+
+    // Start utils region
+
+    @SneakyThrows
```

Comment:
Why is it sneaky throwing here? It does not seem to comply with the guidelines here:

https://github.com/pravega/pravega/wiki/Contributing#exceptions

Reply:
Now we catch and re-throw that exception as a runtime exception.